.cirrus.yml

```diff
@@ -0,0 +1,9 @@
+freebsd_instance:
+  image: freebsd-11-2-release-amd64
+
+test_task:
+  procfs_script: >
+    [ -f /proc/curproc ] || mount -t procfs proc /proc
+  pkg_install_script: pkg install -y bash gawk gmake gsed
```

Comment:
i think we were striving to support BSD make, and stick to POSIX awk and sed over the GNU versions.  what were you seeing that required the GNU versions of either ?

bash makes sense.

Reply:
bsd make complains and I can't build it by just running `make` even on native freebsd vm(non-ci instance). my make-fu is not so good to figure out if it's something trivial or not and I haven't really looked at it.

```make
vagrant@freebsd:~/openrc % make
make: "/home/vagrant/openrc/Makefile" line 14: Cannot open /Makefile.inc
make: "/home/vagrant/openrc/Makefile" line 20: Missing dependency operator
make: "/home/vagrant/openrc/Makefile" line 22: Need an operator
make: "/home/vagrant/openrc/Makefile" line 26: Missing dependency operator
make: "/home/vagrant/openrc/Makefile" line 28: Need an operator
make: "/home/vagrant/openrc/Makefile" line 32: Missing dependency operator
make: "/home/vagrant/openrc/Makefile" line 34: Need an operator
make: "/home/vagrant/openrc/Makefile" line 41: Cannot open /mk/sys.mk
make: "/home/vagrant/openrc/Makefile" line 42: Cannot open /mk/os.mk
make: "/home/vagrant/openrc/Makefile" line 43: Cannot open /mk/subdir.mk
make: "/home/vagrant/openrc/Makefile" line 44: Cannot open /mk/dist.mk
make: "/home/vagrant/openrc/Makefile" line 45: Cannot open /mk/gitver.mk
make: "/home/vagrant/openrc/Makefile" line 48: Missing dependency operator
make: "/home/vagrant/openrc/Makefile" line 50: Missing dependency operator
make: "/home/vagrant/openrc/Makefile" line 52: Need an operator
make: Fatal errors encountered -- cannot continue
make: stopped in /home/vagrant/openrc
```

sed clearly complains about guiisms
check here: https://github.com/OpenRC/openrc/runs/38170947
```
sed: 1: "/^\n*$/{$q1;N;ba
": extra characters at the end of q command
```
awk seems to be fine, I just added it for safety but it's not used here as it has to be called as `gawk`

what's the preferred way of handling bsdsed<->gsed differences?
I see 2 ways:
1* rewrite sed expressions to be agnostic
2* introduce implicit dependency on gsed for tests on bsd and handle sed-gsed somewhere in the scripts, probably using `SED` variable like many projects do.
can't just call `gsed` because gentoo does not install `gsed` symlink or binary for example but some others distros do, but it's inconsistent.

Reply:
nope
gnu
```
q [exit-code]
              Immediately quit the sed script without processing any more input, except that if auto-print is not disabled the current pattern space will be printed.  The exit code argument is a GNU extension.
```

bsd
```
[1addr]q
	     Branch to the end of the script and quit without starting a new
	     cycle.
```

so removing the **1** after **q** makes bsdsed happy but probably will never reach `|| echo` part as there will be no non-zero return code.

Reply:
opened bsdmake issue here https://github.com/OpenRC/openrc/issues/266

Reply:
ok, lets file a bug for the GNU sed thing.  that test logic is way more complicated than it needs to be.  we should just delete the sed entirely.